File: android/app/src/main/java/com/smartexpense/android/domain/usecase/expense/CreateExpenseUseCase.java
package com.smartexpense.android.domain.usecase.expense;

import com.smartexpense.android.data.remote.dto.request.ExpenseRequestDto;
import com.smartexpense.android.data.remote.dto.response.ExpenseResponseDto;
import com.smartexpense.android.domain.repository.ExpenseRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

public class CreateExpenseUseCase {
    private final ExpenseRepository repository;

    public CreateExpenseUseCase(ExpenseRepository repository) {
        this.repository = repository;
    }

    public void execute(double amount, String category, String photoUrl, String caption, String expenseDate, ResultCallback<ExpenseResponseDto> callback) {
        ExpenseRequestDto request = new ExpenseRequestDto(amount, category, photoUrl, caption, expenseDate);
        repository.createExpense(request, callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/expense/CreateExpenseUseCase.kt
package com.smartexpense.android.domain.usecase.expense

import com.smartexpense.android.data.remote.dto.request.ExpenseRequestDto
import com.smartexpense.android.data.remote.dto.response.ExpenseResponseDto
import com.smartexpense.android.domain.repository.ExpenseRepository

class CreateExpenseUseCase(private val repository: ExpenseRepository) {
    suspend fun execute(amount: Double, category: String, photoUrl: String, caption: String, expenseDate: String): Result<ExpenseResponseDto> {
        val request = ExpenseRequestDto(amount, category, photoUrl, caption, expenseDate)
        return repository.createExpense(request)
    }
}
